GitCommands/Git/GitRevision.cs

```diff
@@ -84,9 +84,10 @@ public static string ToShortSha(String sha)
         {
             if (sha == null)
                 throw new ArgumentNullException("sha");
-            if (sha.Length > 8)
+            const int maxShaLength = 10;
```

Comment:
Why 10? I prefer 8.

Reply:
8 works for many repos, also for GE. For large repos like Linux kernel even 10 are too few.
See the following:
https://blog.cuviper.com/2013/11/10/how-short-can-git-abbreviate/

For GE
git rev-list --all --abbrev=0 --abbrev-commit |
>   awk '{ a[length] += 1 } END { for (len in a) print len, a[len] }'
4 4218
5 11700
6 1390
7 89
8 1

A follow-up could be to use core.abbrev instead of hardcoding the length (then 7 will often be used)

Reply:
we already had a bug related to length change https://github.com/gitextensions/gitextensions/issues/4028